### PR TITLE
feat: inform the sender when the receiver self deleting timer expires

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- Once the message expires on the sender side and the receiver does not see it yet we want to show the information to the user. 

- On one to one conversation we include the user name in the information label
- On gorup conversation we inform the user that is has to be only one participants seeing the message

### Solutions

- Introduce EphemeralMessageExpiredLabel
- Pass ConversationDetailsData to MessageItem to get the information about the conversation type 

Needs releases with:

(https://github.com/wireapp/kalium/pull/1661)
#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

https://user-images.githubusercontent.com/15973138/234568888-c49e6a41-2e1c-448a-949e-d8a7f134cdda.mp4

Uploading group.mp4…

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1714/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1714/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1714/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1714/4/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageExpiration.kt [hunk 1]
```diff
@@ -202,6 +203,7 @@ class SelfDeletionTimer(private val context: Context) {
 
             fun decreaseTimeLeft(interval: Duration) {
                 if (timeLeft.inWholeSeconds != 0L) timeLeft -= interval
+                Log.d("TEST", "time left ${timeLeft.inWholeSeconds}")
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageExpiration.kt]: Should this log be here? If so could we change the `Log` to `appLogger` and add the class tag?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageExpiration.kt [hunk 2]
```diff
@@ -218,21 +220,25 @@ class SelfDeletionTimer(private val context: Context) {
              * alpha value is equal to the ratio = 1 / 2.5 = 0.4
              */
             fun alphaBackgroundColor(): Float {
-                val timeElapsed = expireAfter - timeLeft
-                val timeElapsedRatio = timeElapsed / expireAfter
+                return if (timeLeft != ZERO) {
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageExpiration.kt]: Could there be any possibility that `timeLeft` is lower than `0`? if so or maybe just to be on the safe side we could change `!=` to `timeLeft > ZERO`?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageExpiration.kt (hunk 1 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageExpiration.kt]: No definetly left over from testing thank u

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageExpiration.kt (hunk 2 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageExpiration.kt]: Not really possible as we normalize it on Kalium side, but worth adding it thanks <3

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1714/5/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt [hunk 3]
```diff
@@ -202,28 +220,45 @@ fun MessageItem(
...
         }
     }
 }
 
+@Composable
+fun EphemeralMessageExpiredLabel(conversationDetailsData: ConversationDetailsData) {
+    val stringResource = if (conversationDetailsData is ConversationDetailsData.OneOne) {
+        stringResource(
+            R.string.label_information_waiting_for_recipent_timer_to_expire_one_to_one,
+            conversationDetailsData.otherUserName ?: ""
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt]: Couldn't we use a default string, something like: `Unknown` instead of the empty string?

**Review by alexandreferris — COMMENTED** [review]
Looking good! I just have one more question/comment

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt (hunk 3 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt]: good point, I wonder if we will ever have situation when the other user name could be null, but thanks for pointing this out

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt [hunk 4]
```diff
@@ -106,9 +112,12 @@ fun MessageItem(
             )
         }
 
+        // in case the message is marked as deleted and is expirable, it means that the message
+        // is belonging to the sender and we are waiting for the receiver timer to expire to permanently delete it
+        // we also do not want any background color
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt]: I find this text a bit confusing. Would the following be slightly more clear?:
```suggestion
        // If a message is an expiring one and not deleted yet, it means that the current user is the receiver of it.
        // Once the receiver timer finishes, the message will be deleted locally, and send a signal to the sender so
        // he deletes the message locally as well. In this case, we don't want any background color either.
```

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt [hunk 5]
```diff
@@ -202,28 +220,45 @@ fun MessageItem(
...
                             MessageDecryptionFailure(
                                 messageHeader = header,
                                 decryptionStatus = header.messageStatus as MessageStatus.DecryptionFailure,
                                 onResetSessionClicked = onResetSessionClicked
                             )
                         }
+                        MessageFooter(
+                            messageFooter,
+                            onReactionClicked
+                        )
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt]: Why did you move this?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt [hunk 6]
```diff
@@ -202,28 +220,45 @@ fun MessageItem(
...
+            R.string.label_information_waiting_for_recipent_timer_to_expire_one_to_one,
+            conversationDetailsData.otherUserName ?: ""
+        )
+    } else {
+        stringResource(R.string.label_information_waiting_for_recipent_timer_to_expire_group)
+    }
+
+    Text(
+        text = stringResource,
+        style = typography().body05
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt:259]: We should also make sure that the text color for the gibberish is changed to Gray-70, as it is different from the text color of the normal plain text.

**Review by gongracr — APPROVED** [review]
Great job 💪 ! Just some small comments about code documentation and styles, but preapproving as I assume they are of trivial execution.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt (hunk 4 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt]: this is not really true 🤔 I moved this comment and adjusted it a bit to line 182, pls take a look it it makes more sense to you now

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt (hunk 5 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt]: good catch , thank you

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt (hunk 6 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt:259]: good catch ! thanks

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1714/6/) **failed**.

**Review by alexandreferris — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt (hunk 4 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt]: changed and resolved on the call

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1714/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1714/8/) **failed**.

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-1714/9/) **failed**.

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1714/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6422002)(6422002)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6422002)(6422002)-staging-compat.apk)
